11/headful/al2022/Dockerfile

```diff
@@ -2,9 +2,18 @@ FROM amazonlinux:2022
 
 ARG version=11.0.17.8-1
 RUN set -eux \
-    && yum install -y https://corretto.aws/downloads/resources/$(echo $version | tr '-' '.')/java-11-amazon-corretto-headless-$version.amzn2022.$(uname -m).rpm https://corretto.aws/downloads/resources/$(echo $version | tr '-' '.')/java-11-amazon-corretto-$version.amzn2022.$(uname -m).rpm \
+    && rpm --import file:///etc/pki/rpm-gpg/RPM-GPG-KEY-amazon-linux-2022 \
+    && echo "localpkg_gpgcheck=1" >> /etc/dnf/dnf.conf \
+    && dnf install -y dnf-plugins-core \
+    && dnf download --downloaddir=/var/cache/dnf/corretto https://corretto.aws/downloads/resources/$(echo $version | tr '-' '.')/java-11-amazon-corretto-headless-$version.amzn2022.$(uname -m).rpm https://corretto.aws/downloads/resources/$(echo $version | tr '-' '.')/java-11-amazon-corretto-$version.amzn2022.$(uname -m).rpm \
+    # This will be the count of packages with valid signatures, since the only signatures are from the AL2022 repos this verifies that the Corretto packages are from a trusted source.
+    && VERIFIED_COUNT=$(rpm -K /var/cache/dnf/corretto/*.rpm | grep -c ": digests signatures OK$") \
+    && [ 2 -eq ${VERIFIED_COUNT} ] \
```

Comment:
This `2 -eq ...` bit seems really fiddly -- wouldn't it be better to do something like `for rpm in /var/cache/dnf/corretto/*.rpm; do rpm -K "$rpm" || exit 1; done` ? (Does `rpm -K` have suitable exit codes?)

Then we should be safe to `dnf install -y` on the same glob afterwards, right?

Reply:
rpm does not have suitable exit codes which is why I went with grep. Let me see if the curl works and post a new rev.